src/core/server/plugins/plugins_config.ts

```diff
@@ -40,8 +40,16 @@ export class PluginsConfig {
    */
   public readonly pluginSearchPaths: ReadonlyArray<string>;
 
+  /**
+   * Defines directories where a known plugin exists.
+   */
+  public readonly knownPluginPaths: ReadonlyArray<string>;
```

Comment:
__nit:__ hmm, maybe that's just me, but `known` feels a bit confusing, `pluginSearchPaths` is technically also `kwown` :) What about `extraPluginPaths` or `additionalPluginPaths` to stress that these are not our "main" plugin paths, but something that is extra (for dev/test purposes).